src/EditNickname/utils/handleCheckInput.ts

```diff
@@ -0,0 +1,36 @@
+import { CheckNicknameProps } from '../types/editNicknameTypes';
+import checkInputRange from './checkInputRange';
+
+/** 8자 이하 & 한글, 영어, 숫자만 입력 가능하도록 & 첫번째 글자는 공백 불가능 체크 함수*/
+const handleChangeInput = (props: CheckNicknameProps) => {
+  const {
+    handleSetNickname,
+    handleSetWordCnt,
+    handleSetIsValid,
+    currentNickname,
+    e,
+  } = props;
+
+  const input = e.target.value;
+
+  if (e.target.value.length <= 8 && checkInputRange(input[input.length - 1])) {
+    if (e.target.value === ' ') {
+      handleSetNickname('');
+      handleSetWordCnt(0);
+    } else {
+      handleSetNickname(e.target.value);
+      handleSetWordCnt(e.target.value.length);
+    }
+    if (e.target.value === currentNickname) {
+      handleSetIsValid('unchanged');
+    } else {
+      handleSetIsValid('valid');
+    }
```

Comment:
P5: 개인적인 취향이지만..! 조건에 따라 실행되는 코드의 양이 많지 않다면 아래 코드처럼 구현하는게 더 깔끔한 것 같기도 하네용 !
물론 완전히 개인적인 의견이니까 꼭 반영하지 않아도 됩니당 !!
```suggestion
   e.target.value === currentNickname ? handleSetIsValid('unchanged') : handleSetIsValid('valid');
```

Reply:
동감합니다...... 바로 반영해둘게요!!!